### PR TITLE
Fork dialogs with goals - Chapter 2

### PR DESCRIPTION
**Pull request by smashthestate**
- Combine `forkAny` and `neverComplete `to fork certain dialogs together with related goals and their actions for a more fluid player experience in Chapter 2
- remove `dialogFinished` bool from `modifier_dk_last_hit_chapter2_creeps`

Todo:
- Script_2_Courier_3 text is a bit confusing, think we should separate it into 2 lines, one that talks about the two shops and the second one mentioning how to move over to the Radiant one. By forking the current line I didn't see a way to adjust our minimap markers in a good way, without forking the sequence is a bit less fluid

### CODE REVIEW

**Comment by RobinKa** [comment]
Played it and looked fine

File: game/scripts/vscripts/Sections/Chapter2/SectionCreeps.ts [hunk 1]
```diff
@@ -116,12 +115,15 @@ const onStart = (complete: () => void) => {
                     tg.neverComplete()
                 ]),
                 tg.seq([
-                    tg.setCameraTarget(radiantCreeps[0]),
-                    tg.wait(3.5),
+                    tg.wait(2.5),
```
Comment (Shushishtok) [inline game/scripts/vscripts/Sections/Chapter2/SectionCreeps.ts:118]: Is this change necessary? 3.5 is more or less the time it takes to the creeps to meet and start fighting.

File: game/scripts/vscripts/Sections/Chapter2/sectionCourier.ts [hunk 2]
```diff
@@ -94,27 +94,38 @@ const onStart = (complete: () => void) => {
...
-            freezePlayerHero(false)
-        }),
-        tg.goToLocation(inFrontOfRadiantSecretShopLocation, _ => [inFrontOfTheRiverLocation, insideRiverLocation, upSecretShopRamp]),
+
+        // Fork secret shop waypoint dialog
+        tg.forkAny([
+            tg.seq([
+                tg.audioDialog(LocalizationKey.Script_2_Courier_3, LocalizationKey.Script_2_Courier_3, context => context[CustomNpcKeys.SlacksMudGolem]),
+                // tg.immediate(context => {
+                //     MinimapEvent(DotaTeam.GOODGUYS, context[CustomNpcKeys.SlacksMudGolem], radiantSecretShopLocation.x, radiantSecretShopLocation.y, MinimapEventType.CANCEL_TELEPORTING, 5)
```
Comment (Shushishtok) [inline game/scripts/vscripts/Sections/Chapter2/sectionCourier.ts:110]: Is this intended to be commented out?

File: game/scripts/vscripts/Sections/Chapter2/sectionCourier.ts [hunk 3]
```diff
@@ -94,27 +94,38 @@ const onStart = (complete: () => void) => {
...
+                //     MinimapEvent(DotaTeam.GOODGUYS, context[CustomNpcKeys.SunsFanMudGolem], direSecretShopLocation.x, direSecretShopLocation.y, MinimapEventType.CANCEL_TELEPORTING, 5)
+                // }),
+                tg.neverComplete()
+            ]),
+            tg.seq([
+                tg.immediate(() => {
+                    goalMoveToSecretShop.start()
```
Comment (Shushishtok) [inline game/scripts/vscripts/Sections/Chapter2/sectionCourier.ts:117]: I would move the goal start to be after the 3 seconds wait.

File: game/scripts/vscripts/Sections/Chapter2/sectionCourier.ts [hunk 4]
```diff
@@ -304,6 +348,11 @@ export function chapter2CourierOrderFilter(event: ExecuteOrderFilterEvent): bool
                     return true
                 }
             }
+        } else if (units.length == 1) {
```
Comment (Shushishtok) [inline game/scripts/vscripts/Sections/Chapter2/sectionCourier.ts:351]: ===

**Review by Shushishtok — CHANGES_REQUESTED** [review]

File: game/scripts/vscripts/Sections/Chapter2/SectionCreeps.ts (hunk 1 above)
Comment (smashthestate) [reply game/scripts/vscripts/Sections/Chapter2/SectionCreeps.ts]: The panCamera step that I added takes about a second or so, so with this adjustment the dialogue syncs well with the creeps connecting

File: game/scripts/vscripts/Sections/Chapter2/sectionCourier.ts (hunk 2 above)
Comment (smashthestate) [reply game/scripts/vscripts/Sections/Chapter2/sectionCourier.ts]: My mistake, was testing to see if I can remove it and forgot to do it after. Gone now.

File: game/scripts/vscripts/Sections/Chapter2/sectionCourier.ts (hunk 3 above)
Comment (smashthestate) [reply game/scripts/vscripts/Sections/Chapter2/sectionCourier.ts:113]: Yes that's better, done.

**Review by Shushishtok — APPROVED** [review]
